test/model/src/core/kv_table_item.cpp

```diff
@@ -171,74 +172,117 @@ kv_table_item::contains_any(const data_value &value, timestamp_t timestamp)
     return false;
 }
 
+/*
+ * kv_table_item::exists --
+ *     Check whether the latest value exists.
+ */
+bool
+kv_table_item::exists()
+{
+    data_value out;
+    int ret = get(k_timestamp_latest, out);
+    if (ret != 0 && ret != WT_NOTFOUND) {
+        std::ostringstream error;
+        error << "Get returned: " << ret;
+        throw model_exception(error.str());
+    }
+    return ret != WT_NOTFOUND;
+}
+
 /*
  * kv_table_item::get --
- *     Get the corresponding value. Note that this returns a copy of the object.
+ *     Get the corresponding value.
  */
-data_value
-kv_table_item::get(timestamp_t timestamp)
+int
+kv_table_item::get(timestamp_t timestamp, data_value &out)
 {
     std::lock_guard lock_guard(_lock);
     kv_update::timestamp_comparator cmp;
 
-    if (_updates.empty())
-        return NONE;
+    if (_updates.empty()) {
+        out = NONE;
+        return WT_NOTFOUND;
+    }
+
+    if (has_prepared_nolock(timestamp)) {
+        out = NONE;
+        return WT_PREPARE_CONFLICT;
+    }
 
     auto i = std::upper_bound(_updates.begin(), _updates.end(), timestamp, cmp);
-    if (i == _updates.begin())
-        return NONE;
-    return (*(--i))->value();
+    if (i == _updates.begin()) {
+        out = NONE;
+        return WT_NOTFOUND;
+    }
+
+    out = (*(--i))->value();
+    return out == NONE ? WT_NOTFOUND : 0;
 }
 
 /*
  * kv_table_item::get --
- *     Get the corresponding value. Note that this returns a copy of the object.
+ *     Get the corresponding value.
  */
-data_value
-kv_table_item::get(kv_transaction_ptr txn)
+int
+kv_table_item::get(kv_transaction_ptr txn, data_value &out)
```

Comment:
I feel like this was discussed in the previous PR, but the proliferation of out parameters here makes me think `std::optional` is worth considering.

Reply:
I'm not sure how this would help. Can you please elaborate? `data_value` already has a `NONE` value, which was returned for when the value was deleted, but I needed to make this change as reading can now return WiredTiger errors.

If we don't like the `out` argument, I can think of two options:
* Return an instance of `std::variant<int, model::data_value>`, which can be used to represent either an error or a valid result. I decided against it, as returning values via parameters is a lot more common in C/C++; I've only really seen this alternative in Go, and I don't think it's particularly common in our code, so I didn't want to unnecessarily introduce a new non-standard way of doing it.
* Throw an exception, so that we can continue to return a `data_value`. This would be more C++ like, but I ended up not doing this, so that the top-level function can return a WiredTiger compatible return code.

If there is desire to remove the `out` arguments, I can switch to the latter approach with exceptions, and only translate them to WiredTiger-compatible int return codes at the top-level `kv_table` API, so that they can be compared directly to what's returned from regular WiredTiger function calls.

Reply:
After thinking about this more, I ended up switching to throwing exceptions on error, so that we can keep returning `data_value`.

Reply:
Thanks - the code reads better now, at least to me.